File: solutions/python/run-length-encoding/1/run_length_encoding.py
def decode(string):
    res = ""
    for posicion,letra in enumerate(string):
        if letra.isnumeric():
            if string[posicion+1].isnumeric(): 
                res += ((int(string[posicion])*10)*(string[posicion+2])) 
            else:
                res += (int(letra)-1)*string[posicion+1]
                
        else:
            res += letra
    return res
def encode(string):
    cant = 1
    res = ""
    for posicion,letra in enumerate(string):
        if posicion < len(string)-1 and letra == string[posicion+1]:
            cant +=1
        else:
            if cant > 1:
                res += str(cant)
                cant = 1
            res +=letra

    return res

File: solutions/javascript/run-length-encoding/1/run-length-encoding.js
//
// This is only a SKELETON file for the 'Run Length Encoding' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const encode = (text) => {
  let result = "";
  let count = 1;
  for (let i = 0; i < text.length; i++) {
    if (text[i] !== text[i + 1]) {
      if (count > 1) {
        result += count;
      }
      result += text[i];
      count = 1;
    } else count++;
  }
  return result;
};

export const decode = (text) => {
  let res = "";
  let n = "";
  for (let i = 0; i < text.length; i++) {
    if (/[0-9]/.test(text[i])) {
      n += String(text[i]);
    } else {
      res += text[i];
      for (let j = 1; j < Number(n); j++) {
        res += text[i];
      }
      n = "";
    }
  }
  return res;
};
